.github/workflows/ci.yml

```diff
@@ -0,0 +1,67 @@
+name: Micro-RDK CI
+
+on:
+  workflow_call:
+  workflow_dispatch:
+  pull_request:
+    branches: ['main']
+  push:
+    branches:
+      - '**'
+    tags-ignore:
+      - '**'
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+
+jobs:
```

Comment:
Personally, I'm not fond of this. I realize the savings of not doing a build at all are pretty big, but the risk of the filter becoming subtly incorrect outweigh it in my opinion. I'd rather iteratively get caching good enough that we don't care if we do an esp32 build, even if we didn't really change anything esp32 specific.

Reply:
i agree I considered it but you convinced me to remove it